src/model/MintInfo.ts

```diff
@@ -0,0 +1,90 @@
+import { GetInfoResponse, MPPMethod, SwapMethod, WebSocketSupport } from './types';
+
+export class MintInfo {
+	private readonly mintInfo: GetInfoResponse;
+
+	constructor(info: GetInfoResponse) {
+		this.mintInfo = info;
+	}
+
+	isSupported(num: 4 | 5): { disabled: boolean; params: Array<SwapMethod> };
+	isSupported(num: 7 | 8 | 9 | 10 | 11 | 12 | 14): { supported: boolean };
+	isSupported(num: 17): { supported: boolean; params?: Array<WebSocketSupport> };
+	isSupported(num: 15): { supported: boolean; params?: Array<MPPMethod> };
+	isSupported(num: number) {
+		switch (num) {
```

Comment:
Should the cases be handled exhaustively so that unrecognized NUT numbers always resolve to false?
Or alternatively throw on unrecognized NUT number.

Reply:
Very good point! I added a default case that throws. Throwing in this case makes more sense to me, because it makes clear that this NUT is unsupported by the client, not the mint

Reply:
In [nutshell](https://github.com/cashubtc/nutshell/blob/901b16713cd962f613367acd0c77ffa46c42ba2d/cashu/wallet/mint_info.py#L25) there is a general function to check if a mint supports a NUT number, maybe this would be also good here.

Reply:
I have now added a getter for the `nuts` object. That way, people can check whatever they want on the `nuts` object and use the `isSupported` method for all nuts that are implemented in cashu-ts